vision/module/location.py

```diff
@@ -34,22 +34,45 @@ def __init__(self):
         self.lower_bound = np.array(0)  # lower bound of slopes
         self.num_buckets = np.array(0)  # number of buckets applied to slopes
 
-    ## Finding Distance to Module
+        self.needsRecalc = (
+            True  # Prevents recalculation of circles, slopes, and slope grouping
+        )
+
+    ## Determining if Module is in frame
 
-    def getDistance(self):
+    def isInFrame(self) -> bool:
         """
-        Finds the distance to the module.
+        Determines if the Module is in the frame
 
         Returns
         -------
-        int - distance to the module.
+        bool - true if module is in the frame and false if module is not in the frame
         """
-        self.distance = self.depth[self.center[1], self.center[0]]
-        return self.distance
+        MIN_SLOPES_IN_BUCKET = (
+            15  # Minimum number of slopes per bucket to identify the module
+        )
+        MAX_CIRCLES = 100  # maximum number of circles that are allowed to be detected before in_frame fails
+
+        if self.needsRecalc:
+            # Circle Detection
+            self._circleDetection()
+
+        if (
+            self.circles is None or self.circles.shape[0] > MAX_CIRCLES
```

Comment:
> self.circles is None

This should be changed from is None to a constant called MIN_CIRCLES that would be set to 4 since there is also no point to calculations if there is 1, 2, or 3 circles.